PeerioTechnologies/peerio-icebear
# feat: limit size of temp cache folder

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7067
https://app.clubhouse.io/peerio/story/277/desktop-limit-file-system-cache

#### Linked PRs in other repos
desktop https://github.com/PeerioTechnologies/peerio-desktop/pull/167
#### Breaking changes  
n/a
#### Manual testing instructions  
On this branch this app should:
1. (unchanged) Download inline images to temp cache folder
2. When folder is over an arbitrary size limit, delete temp images (oldest first) until folder is below that limit, *except* -
3. Never delete temp images that are in the currently selected chat

#### Additional notes
n/a

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Are there any bicycles invented that should be replaced with existing code or external libs?
* Are there unit and e2e tests added?

----
### Repository owner

- [ ] Breaking changes are applied correctly.
